### ISSUE TITLE
Finalize data landscape architecture

### ISSUE THREAD

**Issue by csa-goose** [Police-Data-Accessibility-Project/planning#7]
Structured, semi-structured, RDBMS or flat files... 

Consider layers of the data landscape: ingestion, processing, and serving.

I propose a flat file data lake for raw data. The data will likely have different schemas and therefore raw data should be curated in the processing layer by an ELT/ETL tool. Finally, data should be stored in a performant warehouse for serving.

**Comment by sidsenthil** [comment]
Items should be touched on in planning meeting:

1. Quick introductions (dependent on number of people who show up)
2. Incoming data streams
- Content of data (fields and types)
- Speed and size of data
- Delivery medium (integration with scraping layer)
3. High-level data architecture
- Consensus on roughly 3-tiered architecture? (data lake for raw data ingest, ETL transformations, and data warehouse for modeling and delivery)
4. Technology
- Cloud hosting
- Tools (db, etl, orchestration, etc)
5. Who pays for this stuff?

I'm thinking we keep this relatively open and just see where the wind takes us, unless someone who is opinionated on open-source collaboration has objections.

**Comment by csa-goose** [comment]
I will post 2 times to host a meeting this weekend for an introductory discussion. Please thumbs up or down to vote on which time works best for you.

**Comment by csa-goose** [comment]
Saturday, May 30th at 11:00 EDT

**Comment by csa-goose** [comment]
Sunday, May 31 7:00 PM EDT

**Comment by PoliceWithData** [comment]
@csa-goose flat files for raw data is the best way to go. I would propose storing the scraped HTML files too. This will increase the storage requirements but it would allow us to revisit the files, change how we process the data and even extract more information if possible. 

Cloud hosting: I am in the owned server with self-hosted in a colo camp.

**Comment by jlintag** [comment]
What's the level of effort in structuring towards something cloud agnostic? Seeing as how things are currently amorphous but we also want to maintain momentum, that feels like the right thing to do vs making decisions that make it hard to be flexible in the future (esp cost wise)

**Comment by csa-goose** [comment]
@jlintag - I fully agree this will be a cloud implementation contrary to what PoliceWithData suggested. Though those details can be ironed out and discussed openly on the call Sunday. However, as far as being cloud agnostic, at this stage that is not a major concern. We are simply discussing landing the raw data which could happen anywhere really. The greater vision of the project will guide our requirements.

**Comment by nfrostdev** [comment]
If we could do both that would be great, but I'm unsure of cost implications.

**Comment by csa-goose** [comment]
The Discord server for the call Sunday is a pinned message on the slack/database-discussion channel

**Comment by csa-goose** [comment]
From the management meeting that occurred 5/30:

> Our product is a consolidated, **publicly available** library of law enforcement related **records**. Our goal is to ease data consumption, **facilitate open source software analytics**, and **provide an interface for reporting and analysis** of law enforcement activity. Our product will be made publicly available, free of charge.

**Comment by samcorzine** [comment]
![Untitled Diagram](https://user-images.githubusercontent.com/24443375/83463353-58e65980-a433-11ea-91a0-b7777bfb627c.png)

**Comment by SteveWallace** [comment]
You might look at something like https://github.com/covidatlas/li as a base for quickly ramping up scraping efforts.

**Comment by jdukemerratrix** [comment]
I can't make the meeting unfortunately, but wanted to share an idea that travels between the data viz, FOIA, and scraper teams. It could be useful to create a real-time, geographical visualization of this google doc: https://docs.google.com/spreadsheets/d/1yyjYV1BLFuLy32CW66zApuEWDSFrR9Dw9y49MU7dxcQ/edit#gid=0.  If this has already been discussed, is in the works, etc., feel free to disregard. Let me know if it comes up in the meeting and people think it's worth pursuing.

TL;DR - hierarchy of visual maps to give anyone, but especially new volunteers, a clear heading of where they can go immediately to do work on scraping, recon, etc. for the Google Doc.

My thinking is this:

Perception of Problem:
>There are a lot of geographical areas that people are reporting in. Their first instinct will likely be to look locally for acquiring police use of force records.
>Someone else could be working on that same area, but hasn't fully updated the google doc to reflect that.
>Person 2 begins to perform redundant work, time and energy is wasted. Non-zero chance of a volunteer becoming discouraged due to the experience. Possible addition to volunteer churn and all that comes with it.

Proposed Solution:
>Break US out into a hierarchy of maps, with something like Leaflet or Folium, updated directly from Google Doc Progress. Allow for new users to visually drill down by State --> County --> City to see statuses. 
>It's important that these geographic denotations (whether it's shading, symbols, a combination, etc.) have two levels (at least) of complexity conveyed at first glance, in my opinion. 
>At the first level, something simple like a red/yellow/green color indicator could relay whether anyone has even touched the data cells at some point in the Google Doc, let alone is currently working on it. Something else that could go a ways towards preventing redundancy is having individuals claim what parts of the Google Sheet that they're working on and clearly link it to their profile (profile attribute demonstrating what they've claimed, which links directly to a highlighted subsection of the Google Doc if necessary? Google sheet sections highlighted together which link to someone's profile who has claimed it? Both?)
>The second level would likely have to be a matrix of progress on different fronts - Is someone currently developing a scraper? Which sources of data inflows are available (Use of Force, Citations, Court Records)? How accessible are the records (API, Manual portal login, E-mail Formal Requests Only)? Derivative from last question, how often are available records updated (Seattle for example has an API with daily updates). It seems a lot of these are present in the Google Doc categorically, but having worked on research projects before with multiple people doing qual research and sharing a database, up to date progress that was immediately legible was really important for meeting deliverable goals and collaborating on tougher cases (profiling extremists in this example).
>As the project develops further we may learn that certain areas of the Google Doc are harder, on average, to figure out than others and that they should be prioritized first. It may be harder to convey this hierarchy of importance with just a flat document, especially people who may be out of their element at the outset.
